### PR TITLE
add libphonenumber w/ git autoupdate

### LINKED ISSUE #7805: [Request] Add libphonenumber
**Library name:** libphonenumber
**Git repository url:** https://github.com/googlei18n/libphonenumber
**npm package url(optional):** 
**License(s):** Apache License 2.0
**Official homepage:** 
**Wanna say something? Leave message here:**
# 

Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #7805 
@x09326  please help me review this pull request, thank you.
The issue discussed with author is below
https://github.com/googlei18n/libphonenumber/issues/1325, https://github.com/googlei18n/libphonenumber/issues/1362

Manually add other versions and change version name from git repo because the old versions come with unusual prefix.
Update .gitignore to ignore the old version with unusual prefix

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Profile of the lib
- Git repository (required): https://github.com/googlei18n/libphonenumber
- Official website (optional, not the repository):
- NPM package url (optional):
- GitHub / Bitbucket popularity (required):
  - Count of stars: 3973
  - Count of watchers: 205
  - Count of forks: 495
- NPM download stats (optional):
  - Downloads in the last day:
  - Downloads in the last week:
  - Downloads in the last month:
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry
- [ ] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

git auto-update config added, close #7805, cc @lararennie

### CODE REVIEW

File: ajax/libs/libphonenumber/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/googlei18n/libphonenumber.git",
+    "basePath": "javascript/i18n/phonenumbers",
+    "files": [
+      "asyoutypeformatter!(_test*)",
+      "metadata!(fortesting*|lite*)",
+      "phonemetadata.pb*",
+      "phonenumber.pb*",
+      "phonenumberutil!(_test*)"
```
Comment (pvnr0082t) [inline ajax/libs/libphonenumber/package.json]: I think `"+(asyoutypeformatter|metadata|phonemetadata.pb|phonenumber.pb|phonenumberutil).+(js|map)"` would be easier and cleaner. What do you think? @kennynaoh

Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: This is a little hard to read.

Comment (pvnr0082t) [inline ajax/libs/libphonenumber/package.json]: So is the config written by @kennynaoh fine?

Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: If it works, it'll be fine, I didn't test it yet.

**Comment by x09326** [comment]
@kennynaoh 
Please rebase to the latest branch.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by x09326** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
@pvnr0082t @x09326 just to make sure, did you guys test the auto-update config?

**Comment by pvnr0082t** [comment]
Yes, I tried it.

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
I have changed the version name from `libphonenumber-x.x.x` to `x.x.x` and also added the `.gitignore`
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
@kennynaoh You didn't minify the files in old versions.

File: .gitignore [hunk 2]
```diff
@@ -156,6 +156,7 @@ ajax/libs/ng-pdfviewer/r0.2.1/
 ajax/libs/prettyCheckable/.2.1.1/
 ajax/libs/Hyphenator/HyphenatorBeta12/
 ajax/libs/Hyphenator/ersion_*
+ajax/libs/libphonenumber/libphonenumber-*/
```
Comment (pvnr0082t) [inline .gitignore]: @PeterDaveHello If this line is added, I think this library will not automatically update. Is it need to discuss with author to change the tagged scheme?

Comment (PeterDaveHello) [inline .gitignore]: What is the reason why it will not update? All the new versions come with that prefix? If so, yes, I'll like to discuss with its author.

Comment (kennynaoh) [inline .gitignore]: SURE

**Comment by kennynaoh** [comment]
@pvnr0082t , @x09326 
The author published the latest version with non-prefix name.
Please help me review it again.
Thanks!

**Comment by pvnr0082t** [comment]
LGTM

File: ajax/libs/libphonenumber/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "target": "git://github.com/googlei18n/libphonenumber.git",
+    "fileMap": [
+      {
+        "basePath": "javascript/i18n/phonenumbers",
+        "files": [
+          "asyoutypeformatter!(_test*)",
+          "metadata!(fortesting*|lite*)",
+          "phonemetadata.pb*",
+          "phonenumber.pb*",
+          "phonenumberutil!(_test*)"
```
Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: Let's combine to something like `phone*!(_test*).+(js|map)`?

File: ajax/libs/libphonenumber/package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "license": "Apache-2.0",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/googlei18n/libphonenumber.git",
+    "fileMap": [
+      {
+        "basePath": "javascript/i18n/phonenumbers",
+        "files": [
+          "asyoutypeformatter!(_test*)",
+          "metadata!(fortesting*|lite*)",
```
Comment (kennynaoh) [inline ajax/libs/libphonenumber/package.json]: OK, I will add it.

File: ajax/libs/libphonenumber/package.json (hunk 3 above)
Comment (kennynaoh) [inline ajax/libs/libphonenumber/package.json]: This `fileMap` will still grab the file `phonenumberutil_test.js`, maybe we should separate it?

Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: I just say "something like" ...

Comment (kennynaoh) [inline ajax/libs/libphonenumber/package.json]: I think I can combined `phonemetadata.pb*` and `phonenumber.pb*` to `+(phonemetadata.pb*|phonenumber.pb*)`, and the others remain unchanged.
What about this?

Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: The part I want to reuse here is `phone`.

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t I have updated the fileMap, please help me review it again. Thanks!

File: .gitignore [hunk 5]
```diff
@@ -171,6 +171,159 @@ node_modules/
...
+ajax/libs/ng-pdfviewer/r0.2.1/
+ajax/libs/prettyCheckable/.2.1.1/
+ajax/libs/Hyphenator/HyphenatorBeta12/
+ajax/libs/Hyphenator/ersion_*
+ajax/libs/gemma/0.1.0/
+ajax/libs/gemma/0.4.0/
+ajax/libs/origamijs/0.0.2/
+ajax/libs/mobx-react/*-observableprops*/
+ajax/libs/mobx-react/*-fix*/
+ajax/libs/libphonenumber*/
```
Comment (pvnr0082t) [inline .gitignore]: Why did you make so much unrelated changes?

**Comment by kennynaoh** [comment]
@pvnr0082t @x09326 I have updated the `.gitignore` file. Please help me review it. Thanks!

File: .gitignore [hunk 6]
```diff
@@ -171,6 +171,7 @@ node_modules/
 /ajax/libs/react-data-grid/*-*/
 !ajax/libs/react-data-grid/*-alpha*/
 /ajax/libs/react-data-grid/*-dm*/
+/ajax/libs/libphonenumber*/
```
Comment (pvnr0082t) [inline .gitignore]: @kennynaoh Shouldn't it be `/ajax/libs/libphonenumber/libphonenumber*`?

Comment (kennynaoh) [inline .gitignore]: Right, it's my mistake. Please help me review it. Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
Ping @x09326

File: ajax/libs/libphonenumber/package.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/googlei18n/libphonenumber.git",
+    "fileMap": [
+      {
+        "basePath": "javascript/i18n/phonenumbers",
+        "files": [
+          "asyoutypeformatter!(_test*)",
+          "metadata!(fortesting*)",
+          "phone!(*_test*)"
```
Comment (maruilian11) [inline ajax/libs/libphonenumber/package.json]: @kennynaoh could you tell me why use `!(_test*)` for `asyoutypeformatter`, but `!(*_test*)` for `phone`.

Comment (kennynaoh) [inline ajax/libs/libphonenumber/package.json]: It's different case. BTW, `!(test*)` is wrong rule, but I think `!(*test*)` it's ok.

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t @maruilian11 I changed `!(*_test*)` to `!(*test*)`. Please help me review it again. Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
@kennynaoh Did you change to `!(*test*)`? I saw the old version.

File: ajax/libs/libphonenumber/package.json [hunk 8]
```diff
@@ -22,7 +22,7 @@
         "files": [
           "asyoutypeformatter!(_test*)",
           "metadata!(fortesting*)",
-          "phone!(*_test*)"
+          "phone!(*test*)"
```
Comment (maruilian11) [inline ajax/libs/libphonenumber/package.json]: why change it in this commit? not in the first one?

Comment (maruilian11) [inline ajax/libs/libphonenumber/package.json]: btw, I think we can use !(*test*) in 3 rules, it's easy to read, and I tested it.

**Comment by kennynaoh** [comment]
@x09326 , @pvnr0082t , @maruilian11 I have updated the `fileMap`, please help me review it. Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
@kennynaoh I think you did a great job. There is only some things you may notice.
1. 
>The first line of commit message is less then 50 chars, be clean and clear, easy to understand.

I think
`add old versions of libphonenumber and update .gitignore, cc #7805`

`Manually add other versions and change version name from git repo because the old versions come
with unusual prefix.`
`update .gitignore to ignore the old version come with unusual prefix`
can turn to 
`Add libphonenumber old verisons, cc #7805`

`Manually add old versions and change version names because the they come with unusual prefix.`
`Update .gitignore to ignore the old versions with unusual prefix`

2. Don't add v7.7.5 and v7.7.4 to reduce reviewers' loading.
Adding those versions doesn't match the commit msg.

3. not modify the version name, like v7.7.5 -> 7.7.5

Thank you!

**Comment by kennynaoh** [comment]
@maruilian11 Thank you for advice

**Comment by kennynaoh** [comment]
@maruilian11 @x09326 @pvnr0082t I have updated the commit message and removed two versions `v7.7.4` and `v7.7.5`. Please help me review it. Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
@kennynaoh sorry for the late reply.
please rebase this branch with the latest master.
since you need to rebase this branch, can you:
1. change `0.0.8` -> `v0.0.8`
2. let first line of commit message be less then 50 chars

thanks!

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t @maruilian11 My first line of first commit is less than 50 chars and I have rebased to the latest master branch. Please help me review it again. Thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by maruilian11** [comment]
LGTM

File: ajax/libs/libphonenumber/package.json [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/googlei18n/libphonenumber.git",
+    "fileMap": [
+      {
+        "basePath": "javascript/i18n/phonenumbers",
+        "files": [
+          "asyoutypeformatter!(*test*)",
+          "metadata!(*test*)",
+          "phone!(*test*)"
```
Comment (PeterDaveHello) [inline ajax/libs/libphonenumber/package.json]: Can we just use `!(*+(demo|test)*).js` here?

Comment (kennynaoh) [inline ajax/libs/libphonenumber/package.json]: That's great!

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t @maruilian11 I have updated fileMap. Please help me review it. Thanks!

**Comment by kennynaoh** [comment]
@x09326 @pvnr0082t @maruilian11 I separated the commit to make it more clear. Please help me review it. Thanks!

File: .gitignore [hunk 10]
```diff
@@ -187,6 +187,8 @@ node_modules/
 /ajax/libs/react-data-grid/*-dm*/
 /ajax/libs/jquery-easing/1.3.0/
 /ajax/libs/jquery-easing/1.3.1/
+/ajax/libs/libphonenumber/libphonenumber*/
+
```
Comment (pvnr0082t) [inline .gitignore]: unnecessary new line.

**Comment by kennynaoh** [comment]
Sorry for that mistake. @x09326 @pvnr0082t @maruilian11 Please help me review it again. Thanks!

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
Thank you all.
